### PR TITLE
add Catch & Retry types

### PR DESCRIPTION
**Pull request by noxasaxon**
also use jsdoc comment style for info popup in IDEs

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [X] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: dist/types/step_function.d.ts [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+/** https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-choice-state.html  */
 export interface ChoiceState extends BaseStateTerminal {
     Type: "Choice";
     Choices: ChoiceRule[];
     Default?: string;
     InputPath?: string;
     OutputPath?: string;
 }
+/** Shared attributes for all regular States (Wait, Pass) & (Task, Parallel, Map, Interaction)
+/* https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-common-fields.html */
```
Comment (ubaniabalogun) [inline dist/types/step_function.d.ts:28]: If the goal here is JSdoc, I think this multi-line comment should be?
```
/**
 * Shared attributes for all regular States (Wait, Pass) & (Task, Parallel, Map, Interaction)
 * https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-common-fields.html 
 */
```

File: lib/types/step_function.ts [hunk 2]
```diff
@@ -31,8 +31,8 @@ export interface ChoiceState extends BaseStateTerminal {
   OutputPath?: string;
 }
 
-// Shared attributes for all regular States (Wait, Pass) & (Task, Parallel, Map, Interaction)
-// https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-common-fields.html
+/** Shared attributes for all regular States (Wait, Pass) & (Task, Parallel, Map, Interaction)
+/* https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-common-fields.html */
```
Comment (ubaniabalogun) [inline lib/types/step_function.ts:35]: See above comment for multi-line JSDoc

File: lib/types/step_function.ts [hunk 3]
```diff
@@ -80,21 +80,35 @@ export interface TaskState extends BaseStateAction {
...
 }
 
-// https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-choice-state.html#amazon-states-language-choice-state-rules
+export interface Catch {
+  ErrorEquals: string[]; // must have at least 1 string.
+  Next: string;
+  ResultPath?: String;
+}
+
+/**https://docs.aws.amazon.com/step-functions/latest/dg/concepts-error-handling.html#error-handling-retrying-after-an-error */
```
Comment (ubaniabalogun) [inline lib/types/step_function.ts:103]: Looks like we're missing a space here aafter `**`

File: lib/types/step_function.ts (hunk 2 above)
Comment (noxasaxon) [reply lib/types/step_function.ts]: updated comment

File: lib/types/step_function.ts (hunk 3 above)
Comment (noxasaxon) [reply lib/types/step_function.ts]: updated comment

**Review by ubaniabalogun — APPROVED** [review]

**Pull request merged**